vendor-extra/LiberoPageBundle/src/EventListener/BuildView/ItemRefTeaserListener.php

```diff
@@ -0,0 +1,55 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Libero\LiberoPageBundle\EventListener\BuildView;
+
+use FluentDOM;
+use FluentDOM\DOM\Element;
+use GuzzleHttp\ClientInterface;
+use Libero\ViewsBundle\Views\SimplifiedViewConverterListener;
+use Libero\ViewsBundle\Views\TemplateView;
+use Libero\ViewsBundle\Views\View;
+use Libero\ViewsBundle\Views\ViewConverter;
+use Psr\Http\Message\ResponseInterface;
+
+final class ItemRefTeaserListener
+{
+    use SimplifiedViewConverterListener;
+
+    private $client;
+    private $converter;
+
+    public function __construct(ClientInterface $client, ViewConverter $converter)
+    {
+        $this->client = $client;
+        $this->converter = $converter;
+    }
+
+    protected function handle(Element $object, TemplateView $view) : View
+    {
+        return $this->client
+            ->requestAsync(
+                'GET',
+                "{$object->getAttribute('service')}/items/{$object->getAttribute('id')}/versions/latest"
+            )
+            ->then(
+                function (ResponseInterface $response) use ($object, $view) : View {
+                    $item = FluentDOM::load((string) $response->getBody());
+                    $item->namespaces($object->ownerDocument->namespaces());
+
+                    return $this->converter->convert($item->documentElement, $view->getTemplate(), $view->getContext());
+                }
+            )->wait();
```

Comment:
We need to introduce `LazyView`. @thewilkybarkid can we pair on this tomorrow briefly. I have your code sample but I think the master branch has diverged from that fairly significantly.